include/nighthawk/user_defined_output/user_defined_output_plugin.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+
+#include "external/envoy/source/common/common/statusor.h"
+#include "external/envoy/source/common/http/header_map_impl.h"
+
+namespace Nighthawk {
+
+// Information about a Nighthawk worker thread. May expand to contain more fields over time as
+// desired.
+struct WorkerMetadata {
+  // Identifies which worker instantiated the plugin instance.
+  int worker_number;
+};
+
+/**
+ * An interface for the UserDefinedOutputPlugin that receives responses and allows users to
+ * attach their own custom output to each worker Result.
+ *
+ * All UserDefinedOutputPlugins must be thread safe, as it may receive multiple responses
+ * simultaneously. In addition, handleResponseData and handleResponseHeaders may be called in any
```

Comment:
optional nit: s/simultaneously/concurrently/. I think saying concurrently may be more accurate.